### PR TITLE
feat(external-plugins) add guide on testing with js plugins

### PR DESCRIPTION
**Pull request by fffonion**
### Review

@reviewer

### Reason

Add guide on how to test JS plugins

### CODE REVIEW

File: app/enterprise/2.4.x/external-plugins.md [hunk 1]
```diff
@@ -390,6 +390,37 @@ Note in this case, the node version and architecture that runs the plugin server
 the one that runs `npm install` under plugins directory must match. For example, it may break
 when you run `npm install` under macOS and mount the working directory into a Linux container.
 
+### Testing
+
+[kong-js-pdk] provides a mock framework to test plugin code correctness through `jest`.
+
+User will first install `jest` as a development dependency, and add `test` script in package.json:
```
Comment (tlblessing) [inline app/enterprise/2.4.x/external-plugins.md:397]:
```suggestion
Install `jest` as a development dependency, and add  the `test` script in `package.json`:
```

File: app/enterprise/2.4.x/external-plugins.md [hunk 2]
```diff
@@ -390,6 +390,37 @@ Note in this case, the node version and architecture that runs the plugin server
...
+
+```
+npm install jest --save-dev
+```
+
+package.json will have a content similar to following:
```
Comment (tlblessing) [inline app/enterprise/2.4.x/external-plugins.md:403]:
```suggestion
The `package.json` has content similar to the following:
```

File: app/enterprise/2.4.x/external-plugins.md [hunk 3]
```diff
@@ -390,6 +390,37 @@ Note in this case, the node version and architecture that runs the plugin server
...
+    }
+
+Run the test through npm with:
+
+```
+npm test
+```
+
+**Note**: Check out [this repository](https://github.com/Kong/kong-js-pdk/tree/master/examples)
+for example on how to write test using `jest`.
```
Comment (tlblessing) [inline app/enterprise/2.4.x/external-plugins.md:422]:
```suggestion
for examples on how to write test using `jest`.
```

**Review by tlblessing — CHANGES_REQUESTED** [review]
Thank you for the PR @fffonion. Will you be updating the OSS side as well?

**Comment by fffonion** [comment]
@tlblessing Yes that will apply on both sides, I will copy paste the change to OSS.

**Comment by tlblessing** [comment]
Direct review links:

https://deploy-preview-2868--kongdocs.netlify.app/enterprise/2.4.x/external-plugins/#testing

**Review by tlblessing — APPROVED** [review]
Thank you @fffonion !

**Pull request merged**